src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs

```diff
@@ -851,30 +854,42 @@ internal SSHConnection[] ParseSSHConnectionHashTable()
                         throw new PSArgumentException(RemotingErrorIdStrings.InvalidSSHConnectionParameter);
                     }
 
-                    string paramValue = item[paramName] as string;
-                    if (string.IsNullOrEmpty(paramValue))
+                    if (paramName.Equals(PortParameter, StringComparison.OrdinalIgnoreCase))
```

Comment:
Ahh, I did this as we now need to handle InvalidSSHConnectionParameter for value types (integer), wereas previously we would check whether paramValue (as string) was null.

The only reason I added the extra level of if-else was to handle the integer/string values differently, although I do agree that it isn't ideal :(

I'll take another look at this later.